File: Backend/Models/Storageaccount.cs
using System;
using System.Collections.Generic;

namespace SkyVault.Models;

public partial class Storageaccount
{
    public Guid Storageaccountid { get; set; }

    public Guid Providerid { get; set; }

    public string Accountname { get; set; } = null!;

    public long Totalcapacitybytes { get; set; }

    public long Usedcapacitybytes { get; set; }

    public int Priority { get; set; }

    public bool Isactive { get; set; }

    public DateTime Createdat { get; set; }

    public virtual Storageprovider Provider { get; set; } = null!;

    public virtual ICollection<Userfile> Userfiles { get; set; } = new List<Userfile>();
}

File: Frontend/src/models/storageAccount/StorageAccount.ts
export interface CreateStorageAccountRequest {
  providerId: string
  accountName: string
  totalCapacityBytes: number
  priority: number
}

export interface UpdateStorageAccountRequest {
  accountName: string
  totalCapacityBytes: number
  priority: number
}

export interface StorageAccountResponse {
  storageAccountId: string
  providerId: string
  providerName: string
  providerType: string
  accountName: string
  totalCapacityBytes: number
  usedCapacityBytes: number
  availableCapacityBytes: number
  priority: number
  isActive: boolean
  createdAt: string
}
